File: 274-h-index/ken/274.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """

        h = 0
        citations.sort(reverse=True)

        for i in xrange(len(citations)):
            if i + 1 <= citations[i]:
                h = i + 1
            else:
                break

        return h

File: 274-h-index/leochang/274.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int length = citations.size();
        sort(citations.begin(), citations.end());
        
        for (int i = 0; i < length; i++) {
            if (length-i <= citations[i]) {
                return length-i;
            }
        }
        return 0;
    }
};
